project/profiles/capacity4more/modules/c4m/message/c4m_message/c4m_message.module

```diff
@@ -280,6 +513,156 @@ function c4m_message_format_group_owner_public_profile_url(Message $message) {
+
+  return $wrapper->field_node->label();
+}
+
+/**
+ * Resolves the type of attached node.
+ *
+ * @param Message $message
+ *   The Message entity.
+ *
+ * @return string
+ *   Node type.
+ */
+function c4m_message_format_node_type(Message $message) {
+  $wrapper = entity_metadata_wrapper('message', $message);
+
+  return $wrapper->field_node->value()->type;
+}
+
+/**
+ * Resolves the path to attached node.
+ *
+ * @param Message $message
+ *   The Message entity.
+ *
+ * @return string
+ *   Node path.
+ */
+function c4m_message_format_node_url(Message $message) {
+  $wrapper = entity_metadata_wrapper('message', $message);
+
+  return 'node/' . $wrapper->field_node->value(array('identifier' => TRUE));
+}
+
+/**
+ * Resolves number of times attached node was recommended.
+ *
+ * @param Message $message
+ *   The Message entity.
+ *
+ * @return int
+ *   Number of recommendations.
+ */
+function c4m_message_format_node_recommend_count(Message $message) {
+  $wrapper = entity_metadata_wrapper('message', $message);
+  $node = $wrapper->field_node->value();
+
+  $property_info = array(
+    'flag_name' => 'c4m_og_content_recommend',
+  );
+  $count = c4m_helper_entity_entity_get_properties($node, array(), '', 'node', $property_info);
+
+  return empty($count) ? 0 : $count;
+}
+
+/**
+ * Resolves a teaser of nodes body.
+ *
+ * @param Message $message
+ *   The Message entity.
+ *
+ * @return string
+ *   Node body teaser.
+ */
+function c4m_message_format_node_body_teaser(Message $message) {
+  $wrapper = entity_metadata_wrapper('message', $message);
+  $node = $wrapper->field_node->value();
+  $body_teaser = field_view_field('node', $node, 'c4m_body', 'teaser');
+
+  return strip_tags($body_teaser[0]['#markup']);
+}
+
+/**
+ * Resolves a url that is used to un subscribe from following flagged node.
+ *
+ * @param Message $message
+ *   The Message entity.
+ *
+ * @return string
+ *   Un subscribe url.
+ */
+function c4m_message_format_node_unsubscribe_url(Message $message) {
+  return _c4m_message_format_unsubscribe_url($message, 'node');
+}
+
+/**
+ * Helper function.
+ *
+ * Resolves a url that is used to un subscribe from following an entity that is
+ * embedded inside message.
+ *
+ * @param Message $message
+ *   The Message entity.
+ * @param string $flag_suffix
+ *   The suffix of flag that is used to track message entity.
+ *
+ * @return string
+ *   Un subscribe url.
+ */
+function _c4m_message_format_unsubscribe_url(Message $message, $flag_suffix) {
+  $wrapper = entity_metadata_wrapper('message', $message);
+  $entity_type = $flag_suffix == 'taxonomy_term' ? 'taxonomy_term' : 'node';
+  $field = "field_{$entity_type}";
+  $nid = $wrapper->$field->value(array('identifier' => TRUE));
+  $uid = $message->uid;
+  $account = user_load($uid);
+  $hash = entity_metadata_wrapper('user', $account)->c4m_unsubscribe_hash->value();
+
+  return 'user/' . $uid . '/unsubscribe/' . $flag_suffix . '/' . $nid . '/' . $hash;
```

Comment:
@ordavidil 
Per template, we need a simple string:
<a href="!unsubscribe_url">unsubscribe</a>.